Objects/codeobject.c

```diff
@@ -1367,12 +1368,17 @@ deopt_code(_Py_CODEUNIT *instructions, Py_ssize_t len)
 PyObject *
 _PyCode_GetCode(PyCodeObject *co)
 {
+    if (co->_co_code != NULL) {
+        return Py_NewRef(co->_co_code);
+    }
     PyObject *code = PyBytes_FromStringAndSize((const char *)_PyCode_CODE(co),
                                                _PyCode_NBYTES(co));
     if (code == NULL) {
         return NULL;
     }
     deopt_code((_Py_CODEUNIT *)PyBytes_AS_STRING(code), Py_SIZE(co));
+    assert(co->_co_code == NULL);
```

Comment:
I'm not sure this is correct... could the allocation above have triggered a garbage collection that could have ended up populating this member? I'd be more confident just removing this assert and using `Py_XSETREF` or something instead.

Reply:
Frankly, I'm not sure how that would work. But I see your point.

Reply:
``bytes`` objects are not gc tracked (no `PyGC_Head`) so its allocation cannot trigger a gc collection as it uses `PyObject_Malloc` not `PyObject_GC_New`.
 Also this PR is a backport so it would be better to not deviate it from the 3.12 PR else both should be changed.

Reply:
The backport part isn't an issue because I can just fix it on the main branch.

Anyways, I'll wait for Brandt's reply to see what he says.

Reply:
Gentle ping @brandtbucher for your opinion on this please.

Reply:
Sorry, this slipped off of my radar. @kumaraditya303 is correct, so no need to change this.

I think the fact that GC collections may only be triggered only by GC-tracked allocations may be an implementation detail (I thought it was *all* allocations until right now). But we're the implementation, so for us it's just a detail. 🙃